### ISSUE TITLE
[Suggestion] support windows 10 local TTS engine and .txt file to .mp3

### ISSUE THREAD

**Issue by shengdie** [CaffeineAU/TTSAutomate#9]
Hello,
Windows 10 has more local voices than windows 7,  but it seems only for UWP apps (I am not sure).
And could you make it support txt to mp3, like transform all the `.txt` file in a folder to `.mp3`?

**Comment by CaffeineAU** [comment]
Hi shengdie,

Windows 10 supports the Windows Speech API (SAPI) that I currently use for TTSAutomate, and also the Windows RT Speech API.

I'll look at offering the RT Speech API  voices on Windows 10.

For your second suggestion, do you mean convert any txt file in a directory to one mp3 per txt file, or all into one mp3 file?

I'm not sure if I can support that with Ivona voices, as we only get 50000 units of tts per month (a unit is up to 200 characters) and that could potentially burn through that allowance very quickly (we already come close to the limit from regular usage of TTSAutomate)

**Comment by shengdie** [comment]
I mean convert one txt file to one mp3 file, and this may only work with
local TTS engine. Thanks.

2016-11-12 17:14 GMT-05:00 CaffeineAU notifications@github.com:

**Comment by CaffeineAU** [comment]
I've investigated the Windows RT Speech API, and unfortunately it can't be used from within TTSAutomate.
It uses the Windows.Media.SpeechSynthesis namespace, which is only available from a Win RT app, and there's no interoperability available to use anything from within that namespace in .Net / WPF (TTSAutomate). A number of WinRT features can be used from within .Net / WPF, with [some pretty ugly workarounds](http://www.davidmoore.info/blog/2014/10/17/walkthrough-using-winrt-libraries-from-a-windows-desktop-application/), but the SpeechSynthesis namespace isn't one of them.

I built a small test Win RT app to generate some example files, but there are further limitations on where Win RT apps can access files in the filesystem, meaning that it can't access a specific directory unless it's one of the [special subset of allowed directories.](http://lunarfrog.com/blog/winrt-folders-access)

The functionality to convert any text files in a folder to mp3 is interesting, but not part of the core of TTSAutomate. I'll keep it in mind for future developments.

Thanks for the suggestions.
